### PR TITLE
Assignment-7 |( Callbacks, Async/await ,Promises, Promise.all)

### PR DESCRIPTION
**Pull request by kaushalyad321**

### CODE REVIEW

File: Assignment7/problem-3.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+function getPromise(URL) {
+  let promise = new Promise(function (resolve, reject) {
+    let req = new XMLHttpRequest();
+    req.open("GET", URL);
+    req.onload = function () {
+      if (req.status == 200) {
+        resolve(req.response);
+      } else {
+        reject("There is an Error!");
+      }
```
Comment (sumitbaudh22) [inline Assignment7/problem-3.js]: Task 3 not working

File: Assignment7/problem-4.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+// 4. Write a program to implement a Promise-based rate limiter, that limits the number of concurrent requests to a certain number
+
+var limit = require("simple-rate-limiter");
+var request = limit(require("request")).to(10).per(1000);
+userIds.forEach(function(userId) {
+
+	var url = "http://easily-overwhelmed-api.com/users/" + userId;
```
Comment (sumitbaudh22) [inline Assignment7/problem-4.js]: task 4 is not working

File: Assignment7/problem-5.js [hunk 3]
```diff
@@ -0,0 +1,2 @@
+// 5.  Write a program to implement a Promise-based task queue, that processes tasks in a specified order, with a specified concurrency limit
+
```
Comment (sumitbaudh22) [inline Assignment7/problem-5.js]: No code here in this file

Comment (kaushalyad321) [reply Assignment7/problem-5.js]: ok  i left the question due to some doubt related to question

Comment (sumitbaudh22) [reply Assignment7/problem-5.js]: we can discuss those doubts

Comment (kaushalyad321) [reply Assignment7/problem-5.js]: ok

**Pull request merged**